t4c/backup.py

```diff
@@ -140,28 +140,40 @@ def clone_git_repository() -> pathlib.Path:
     git_dir.mkdir(exist_ok=True)
 
     log.debug("Cloning git repository...")
-    subprocess.run(
-        ["git", "clone", os.environ["GIT_REPO_URL"], "/tmp/git"],
-        check=True,
-        env={
-            "GIT_USERNAME": os.getenv("GIT_USERNAME", ""),
-            "GIT_PASSWORD": os.getenv("GIT_PASSWORD", ""),
-            "GIT_ASKPASS": "/etc/git_askpass.py",
-        },
-    )
+    env = {
+        "GIT_USERNAME": os.getenv("GIT_USERNAME", ""),
+        "GIT_PASSWORD": os.getenv("GIT_PASSWORD", ""),
+        "GIT_ASKPASS": "/etc/git_askpass.py",
+    }
+
     try:
         subprocess.run(
-            ["git", "switch", os.environ["GIT_REPO_BRANCH"]],
+            [
+                "git",
+                "clone",
+                "--depth=1",
+                f"--branch={os.environ['GIT_REPO_BRANCH']}",
+                os.environ["GIT_REPO_URL"],
+                "/tmp/git",
+            ],
             check=True,
             cwd=git_dir,
+            env=env,
         )
     except subprocess.CalledProcessError as e:
         print(e.returncode)
         if e.returncode == 128:
             subprocess.run(
-                ["git", "switch", "-c", os.environ["GIT_REPO_BRANCH"]],
+                [
```

Comment:
Have you tested this case? Where do you create the new branch as described in the #209? (For non-existing branches, shallow clone the default branch and create a new one.)

I didn't start the local development environment with T4C server and Git server, but `git push origin branch2` should fail if there the branch `branch2` doesn't exist locally.

Reply:
I tested it with Git locally. It does use the default branch defined on the remote.